parser/main.py

```diff
@@ -188,215 +194,125 @@ def parse_request():
-            return {
-                "result": "INFLUX_WRITE_FAIL",
-                "measurements": extracted_measurements,
-                "id": can_msg.identifier
-            }
+    app.logger.info(f"Received a {message.type} message. ID = {id=}")
 
     return {
         "result": "OK",
-        "measurements": extracted_measurements,
-        "id": can_msg.identifier
+        "message": message.data["display_data"],
+        "id": id,
+        "type": type
     }
 
 
+@app.post(f"{API_PREFIX}/parse/write/debug")
+@auth.login_required
+def parse_and_write_request():
+    return parse_and_write_request_bucket("_test")
+
 @app.post(f"{API_PREFIX}/parse/write/production")
 @auth.login_required
 def parse_and_write_request_to_prod():
-    """
-    Parses incoming request, writes the parsed measurements to InfluxDB production bucket,
-    and sends back parsed measurements back to client.
-    """
-    parse_request = flask.request.json
-    id: str = parse_request["id"]
-    data: str = parse_request["data"]
-    # TODO: use timestamp when writing to Influx
-    timestamp: str = parse_request["timestamp"]
-    data_length: str = parse_request["data_length"]
-
-    app.logger.info(f"Received message: {id=}, {data=}")
+    return parse_and_write_request_bucket("_prod")
 
-    # TODO: add validation for received JSON object
+"""
+Parses incoming request, writes the parsed measurements to InfluxDB bucket (debug or production),
+and sends back parsed measurements back to client.
+"""
+def parse_and_write_request_bucket(bucket):
+    parse_request = flask.request.json
+    message = create_message(parse_request["message"], format_specifier_list)
+    id = message.data.get("ID", "UNKNOWN")
+    type = message.type
 
-    can_msg = StandardFrame(id, data, timestamp, data_length)
+    app.logger.info(f"Received a {message.type} message. ID = {id=}")
 
-    # try extracting measurements from CAN message
+    # try extracting measurements
     try:
-        extracted_measurements: List[Measurement] = can_msg.extract_measurements(CAR_DBC)
-        app.logger.info(f"Successfully parsed CAN message with id={can_msg.hex_identifier}({can_msg.identifier}) and placed into queue")
+        app.logger.info(f"Successfully parsed {type} message with id={id} and placed into queue")
     except Exception:
         app.logger.warn(
-            f"Unable to extract measurements for CAN message with id={can_msg.hex_identifier}({can_msg.identifier})")
+            f"Unable to extract measurements for {type} message with id={id}")
+        app.logger.warn(str(message.data["display_data"]))
         return {
             "result": "PARSE_FAIL",
-            "measurements": [],
-            "id": can_msg.identifier
+            "message": [],
+            "id": id
         }
 
-    # try putting the extracted measurements in the queue for Grafana streaming
-    try:
-        stream_queue.put(extracted_measurements, block=False)
-    except queue.Full:
```

Comment:
Why are we removing the stream queue to Grafana? Have we determined that the latency of Grafana reading from the database is negligible?

Reply:
It was commented for testing the influx db -> grafana pipeline. It'll be uncommented before merging.